File: lib/src/web_workers/shared/render_store.dart
library angular2.src.web_workers.shared.render_store;

import "package:angular2/src/core/di.dart" show Injectable;

@Injectable()
class RenderStore {
  num _nextIndex = 0;
  Map<num, dynamic> _lookupById;
  Map<dynamic, num> _lookupByObject;
  RenderStore() {
    this._lookupById = new Map<num, dynamic>();
    this._lookupByObject = new Map<dynamic, num>();
  }
  num allocateId() {
    return this._nextIndex++;
  }

  void store(dynamic obj, num id) {
    this._lookupById[id] = obj;
    this._lookupByObject[obj] = id;
  }

  void remove(dynamic obj) {
    var index = this._lookupByObject[obj];
    (this._lookupByObject.containsKey(obj) &&
        (this._lookupByObject.remove(obj) != null || true));
    (this._lookupById.containsKey(index) &&
        (this._lookupById.remove(index) != null || true));
  }

  dynamic deserialize(num id) {
    if (id == null) {
      return null;
    }
    if (!this._lookupById.containsKey(id)) {
      return null;
    }
    return this._lookupById[id];
  }

  num serialize(dynamic obj) {
    if (obj == null) {
      return null;
    }
    return this._lookupByObject[obj];
  }
}

File: npm/angular2/es6/prod/src/web_workers/shared/render_store.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Injectable } from "angular2/src/core/di";
export let RenderStore = class RenderStore {
    constructor() {
        this._nextIndex = 0;
        this._lookupById = new Map();
        this._lookupByObject = new Map();
    }
    allocateId() { return this._nextIndex++; }
    store(obj, id) {
        this._lookupById.set(id, obj);
        this._lookupByObject.set(obj, id);
    }
    remove(obj) {
        var index = this._lookupByObject.get(obj);
        this._lookupByObject.delete(obj);
        this._lookupById.delete(index);
    }
    deserialize(id) {
        if (id == null) {
            return null;
        }
        if (!this._lookupById.has(id)) {
            return null;
        }
        return this._lookupById.get(id);
    }
    serialize(obj) {
        if (obj == null) {
            return null;
        }
        return this._lookupByObject.get(obj);
    }
};
RenderStore = __decorate([
    Injectable(), 
    __metadata('design:paramtypes', [])
], RenderStore);
